lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -494,26 +495,33 @@ newTransactionLayer networkId = TransactionLayer
                     mkTx networkId payload ttl stakeCreds keystore wdrl
                         selection delta
 
-    , addVkWitnesses = \_era stakeCreds addressResolver inputResolver sealedTx -> do
+    , addVkWitnesses = \_era usingStakeCreds stakeCreds addressResolver inputResolver sealedTx -> do
```

Comment:
Is it not possible to use the existing `resolveRewardAcct` instead of adding `usingStakeCreds`?

I.e. add a `mapMaybe mkCertWitness certs` in the same style as:

https://github.com/input-output-hk/cardano-wallet/blob/5ad348d26313363bacec061ee963c89fd9d467b8/lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs#L411-L412

I'm worried `usingStakeCreds` makes the responsibility of the function unclear, entangles it with server-handler logic making it difficult to test in the style of Sam's https://github.com/input-output-hk/cardano-wallet/blob/master/lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs#L380-L394

Reply:
ok, so I need to sneak information whether during signing there are OUR delegations in tx. If yes then I need to add additional witness. If not then I don't need to add them, otherwise I will have bloated payload and coin selection give rise to too small fees. Because we have right now, construction and signing decoupled then I need to provide this info either at the server level (and I did it as it was straightforward using decodeTransaction) or I can do it probably at shelley's transaction level. At this stage I just wanted to solve this issue of picking additional witness and make all tests green. But we can try to improve it :+1:  Let me think for a while

Reply:
so that would mean to add those certs during  construcTransaction and then only capture them during signing? :thinking:

Reply:
it seems as redundant argument to signTransaction to have (body,wits). It should be just body and wits lookup in body under certificates

Reply:
In the end I left `(body, wits)` as we will need it for partial signing. `wits` could be wits from other parties...So we need to leave it. Nevertheless, now old tx workflow and new one works, we discover reward account from selaedTx, and translate them into XPrv and pwd when ours. No more artificial sneaking of delegs wits